lib/samples/snap_geometry_edits/snap_geometry_edits.dart

```diff
@@ -0,0 +1,679 @@
+                  const SizedBox(height: 20),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.start,
+                    children: [
+                      Text(
+                        'Select snap sources',
+                        style: Theme.of(context).textTheme.titleLarge,
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: 20),
+                  // Add checkboxes for enabling the point layers as snap sources.
+                  buildSnapSourcesSelection(
+                    'Point layers',
+                    _pointLayerSnapSources,
+                  ),
+                  // Add checkboxes for the polyline layers as snap sources.
+                  buildSnapSourcesSelection(
+                    'Polyline layers',
+                    _polylineLayerSnapSources,
+                  ),
+                  // Add checkboxes for the graphics overlay as snap sources.
+                  buildSnapSourcesSelection(
+                    'Graphics Overlay',
+                    _graphicsOverlaySnapSources,
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget buildSnapSourcesSelection(
+    String label,
+    List<SnapSourceSettings> allSourceSettings,
+  ) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          label,
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+        Column(
+          children: allSourceSettings.map((sourceSetting) {
+            return Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                // Display the layer name, or set default text for graphics overlay.
+                Text(
+                  allSourceSettings == _pointLayerSnapSources ||
+                          allSourceSettings == _polylineLayerSnapSources
+                      ? (sourceSetting.source as FeatureLayer).name
+                      : 'Editor Graphics Overlay',
+                ),
+                // A checkbox to toggle whether this source setting is enabled.
+                Checkbox(
+                  value: sourceSetting.isEnabled,
+                  onChanged: (isEnabled) {
+                    if (isEnabled != null) {
+                      setState(() => sourceSetting.isEnabled = isEnabled);
+                    }
+                  },
+                ),
+              ],
+            );
+          }).toList(),
+        ),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text('Enable all'),
+            // A checkbox to enable all source settings in the category.
+            Checkbox(
+              value: allSourceSettings.firstWhereOrNull(
+                    (snapSourceSettings) =>
+                        snapSourceSettings.isEnabled == false,
```

Comment:
`== false` is kinda weird. This could just be `!snapSourceSettings.isEnabled`

Reply:
Yeah you're right, I often have things more verbose when I'm figuring things out and then refactor. Missed this one, will remove.